storm-server/src/main/java/org/apache/storm/daemon/supervisor/Slot.java

```diff
@@ -104,9 +111,12 @@
             this.port = port;
             this.iSupervisor = iSupervisor;
             this.localState = localState;
+            this.changingCallback = changingCallback;
         }
     }
-    
+
+    //TODO go through all of the state transitions and make sure we handle changingBlobs
+    //TODO make sure to add in transition helpers that clean changingBlobs && pendingChangeingBlobs for not the current topology
```

Comment:
??

Reply:
Yes I left in a lot of TODOs #2363 cleans them up, but I will try to pull back what I can here.

In this particular case I did go through the transitions and I think I have it all covered.